From 6b9438d746cec6e4e632d78c5244f4be6314d1c9 Mon Sep 17 00:00:00 2001
From: Allen Winter <allen.winter@kdab.com>
Date: Sun, 28 May 2017 12:51:10 -0400
Subject: [PATCH] icaltypes.c - icalreqstattype_from_string(), copy the
 reqstattype's debug string into its own memory in the ring buffer.

Issue#253
fix CVE-2016-9584
---
 src/libical/icaltypes.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- libical-r1139/src/libical/icaltypes.c.orig	2020-03-30 13:43:22.256553924 -0500
+++ libical-r1139/src/libical/icaltypes.c	2020-03-30 13:44:06.566690029 -0500
@@ -161,7 +161,7 @@
 
   p2 = strchr(p1+1,';');
   if (p2 != 0 && *p2 != 0){
-    stat.debug = p2+1;
+    stat.debug = icalmemory_tmp_copy(p2 + 1);
   } 
 
   return stat;
